pkg/gateway/sig/v4.go

```diff
@@ -293,6 +358,38 @@ func (ctx *verificationCtx) getAmzDate() (string, error) {
 	return amzDate, nil
 }
 
+func (ctx *verificationCtx) verifyExpiration() error {
+	if !ctx.AuthValue.IsPresigned {
+		// TODO: we currently don't have handling for this
+		return nil
+	}
+	// Get and validate the request timestamp
+	amzDate, err := ctx.getAmzDate()
+	if err != nil {
+		return err
+	}
+
+	requestTime, err := time.Parse(v4timeFormat, amzDate)
+	if err != nil {
+		return err
+	}
+
+	now := time.Now().UTC()
+	timeDiff := now.Sub(requestTime)
+
+	// Check for requests from the future and allow small clock skew
+	if timeDiff < 0 && timeDiff.Abs() > 5*time.Minute {
```

Comment:
Why are we checking for 5min?
What's the behavior in AWS for presigned requests that are less than 5 min in the future?
Please document